src/Microsoft.OData.Core/UriParser/Binders/SelectPathSegmentTokenBinder.cs

```diff
@@ -177,7 +177,7 @@ internal static bool TryBindAsOperation(PathSegmentToken pathToken, IEdmModel mo
                 }
             }
 
-            possibleFunctions = possibleFunctions.EnsureOperationsBoundWithBindingParameter().ToList();
+            possibleFunctions.EnsureOperationsBoundWithBindingParameter();

```

Comment:
Do we really need to validate all of the  possibleFunctions here, or should we just wait and validate at line 199 after we have reduced the number of possible functions based on overloads.